### ISSUE TITLE
vue-cli-plugin-electron-builder plugin proposal

### ISSUE THREAD

**Issue by Akumzy** [nklayman/vue-cli-plugin-electron-builder#519]
**Is your feature request related to a problem? Please describe.**
I have few apps that's built with Electron + Golang, now whenever an update is been made in the Go code I had to manually recompile the Go code and start the Electron process somehow 

**Describe the solution you'd like**
I have forked this repo and bootstrapped working version which can be found here https://github.com/Akumzy/vue-cli-plugin-electron-builder/commit/006de9aad065197a91b88520b7421b7f81e979d9
The plugin authors adds an object with name and script properties
Example:
```js
module.exports = {
  pluginOptions: {
    electronBuilder: {
      plugins: [
        {
          name: 'electron-go',
          script: 'vue-cli-plugin-electron-go/plugin.js'
        }
      ]
    }
  }
}
```
Where `name` is the plugin name and `script` is a the module path to execute 

```js
//vue-cli-plugin-electron-builder/plugin-loader.js
module.exports = async function loadPlugins(api, options = {}) {
  /**@type {{script:string; name:string; options?:any}[]]}*/
  const plugins = options.plugins
  if (Array.isArray(plugins) && plugins.length) {
    for (const plugin of plugins) {
      await require(plugin.script).init(api, options)
    }
    return true
  }
  return false
}
```
The plugins will be invoked before starting the electron process and before bundling the app
which at least solves my use case to recompile the Go code or mount `fs-watcher` on the golang directory the recompile and restart the electron app on file changes.

**Describe alternatives you've considered**
I have considered using npm concurrently package to archive this but restarting the Electron process might not be that straight forward

**Additional context**
![electron-go](https://user-images.githubusercontent.com/25849232/66864273-ab4e3500-ef8c-11e9-9f36-5ae23cf872e0.gif)
`vue-cli-plugin-electron-builder` is also package

**Comment by nklayman** [comment]
Have your tried using the `mainProcessWatch` option? You can point it at a file or folder and the main process will rebuild when that file or folder changes:

```js
// vue.config.js

module.exports = {
  pluginOptions: {
    electronBuilder: {
      // Your main process file will be added by default
      mainProcessWatch: ['src/myGoFile.go', 'src/myGoFolder']
    }
  }
}

**Comment by Akumzy** [comment]
rebuilding Go code with `vue-cli-plugin-electron-builder`?

**Comment by nklayman** [comment]
No, but it would rebuild the main process whenever your go code changes. This way you can set up a watcher to build your go code when it changes, and the main process would be rebuilt whenever the go code is rebuilt.

**Comment by Akumzy** [comment]
Now get what you're trying say I'll give a try and see how it's goes and what that means concurrently will be needed to boot up the Go watcher on before `electron:serve` and `electron: build`

**Comment by Akumzy** [comment]
Ok @nklayman  thanks a lot for the help. I added the development binary path to mainProcessWatch [vue-cli-plugin-electron-go]( https://github.com/Akumzy/vue-cli-plugin-electron-go) work in progress though
